src/docs/codelabs/extended-flex-widget.md

```diff
@@ -0,0 +1,429 @@
+{:.no_toc}
+  Try Dart
+
+  Notice how `Row` contains the `mainAxisSize` property set to `MainAxisSize.max`.
+
+  **1.** Click the **Run** button. 
+
+  **2.** Change the `mainAxisSize` property to `MainAxisSize.min`.
+
+  **3.** Click the **Run** button. 
+{{site.alert.end}}
+{% comment %}
+  Gist: https://gist.github.com/def3c5c3c68459b5bb4bdfed58f5d024
+{% endcomment %}
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=d852e4f07d6c87600fe8e0f186c7a31b&amp;fw=true&amp;split=60" width="100%" height="400px"></iframe>
+
+## mainAxisAlignment property
+
+When `mainAxisSize` is set to `MainAxisSize.max`, `Row` and `Column` may have extra space on their main axes. 
+`mainAxisAlignment` is a property that determines how `Row` and `Column` can position their children *horizontally* in that extra space. `mainAxisAlignment` has six `MainAxisAlignment` enums: 
+
+`MainAxisAlignment.start`
+: Positions children near the beginning of the main axis. (Left for `Row`, top for `Column`)
+
+`MainAxisAlignment.end`
+: Positions children near the end of the main axis. (Right for `Row`, bottom for `Column`)
+
+`MainAxisAlignment.center`
+: Positions children at the middle of the main axis.
+
+`MainAxisAlignment.spaceBetween`
+: Divides the extra space evenly between children.
+
+`MainAxisAlignment.spaceEvenly`
+: Divides the extra space evenly between children and creates an even amount of space before the first child and after the last child.
+
+`MainAxisAlignment.spaceAround`
+: Similar to `MainAxisAlignment.spaceEvenly` but reduces half of the space before the first child and after the last child. 
+
+#### Code example: mainAxisAlignment property
+{:.no_toc}
+{{site.alert.secondary}}
+  Try Dart.
+
+  Notice how `mainAxisAlignment` is set to `MainAxisAlignment.start`.
+
+  **1.** Click the **Run** button.
+
+  **2.** Change the `mainAxisAlignment` property to `MainAxisAlignment.end`.
+
+  **3.** Click the **Run** button. 
+{{site.alert.end}}
+{% comment %}
+  Gist: https://gist.github.com/cb8abed13f90a6a0c7a0ada6f15a09c9
+{% endcomment %}
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=cb8abed13f90a6a0c7a0ada6f15a09c9&amp;fw=true&amp;split=60" width="100%" height="400px"></iframe>
+{{site.alert.tip}}
+  Before moving to the next section, try changing `MainAxisAlignment.start` to one of the other enums. 
+{{site.alert.end}}
+
+## crossAxisAlignment property
+
+Similar to the `mainAxisAlignment` property, 
+`crossAxisAlignment` is a property that determines how `Row` and `Column` can position their children in extra space. 
+However, 
+the `crossAxisAlignment` property determines how `Row` and `Column` can position their children *vertically* in that extra space. 
+Note that most of the `crossAxisAlignment` property's enums only function inside of `Row`. 
+The `crossAxisAlignment` property has five `CrossAxisAlignment` enums:
+
+`CrossAxisAlignment.start`
+: Positions children near the top of the cross axis. (`Row`)
```

Comment:
enums => possible values

(`Row`) => (`Row` only)

Add the "only" on all the relevant lines below..